Overload for string deviceID

stefaneberl opened the pull request:
This has the advantage that deviceIDs are no longer restricted
to be longs. It is possible to pass a 250 character long deviceID,
which can be a UUID or something similar.

stefaneberl committed:
Overload for string deviceID

This has the advantage that deviceIDs are no longer restricted
to be longs. It is possible to pass a 250 character long deviceID,
which can be a UUID or something similar.

y0chanan commented on src/openkit-shared/Protocol/Beacon.cs:
```diff
@@ -159,12 +160,12 @@ public class Beacon
             {
                 SessionNumber = 1;
-                DeviceID = randomNumberGenerator.NextLong(long.MaxValue);
+                DeviceID = randomNumberGenerator.NextLong(long.MaxValue).ToString(CultureInfo.InvariantCulture);
```
this is a line with a lot of stuff happening here. By introducing a linebreak the randomizer part and the string conversion can be split

stefaneberl committed:
Leave old CTOR in base class

To not break ABI compatibility the old CTOR with
long deviceID has to be left.

Additionally remove unused import in BeaconTest and increase readability.

stefaneberl committed:
Merge remote-tracking branch 'origin/master' into feature/device_id_overload

thomasgrassauer commented on src/openkit-shared/Protocol/Beacon.cs:
```diff
@@ -84,7 +85,7 @@ public class Beacon
 
         // max name length
-        private const int MAX_NAME_LEN = 250;
+        internal const int MAX_NAME_LEN = 250;
```
as you touched the const here. you might consider changing the name to comply with the C# naming conventions of constants. this also applies to other constants defined in Beacon.cs

stefaneberl committed:
Rename constants to fit C# naming rules

stefaneberl committed:
Fix compiler erro

stefaneberl committed:
Merge branch 'master' into feature/device_id_overload
